### PR TITLE
UI: Fix new bugs from disable/enable feature

### LINKED ISSUE #536: UI: new bugs
It is not possible to add the permissions global.enableUser and global.disableUser in UI
 Closed resources must be ignored when checking assignments (API level bug)
 After canceling the window to add global permissions, the userlist re-fetches, this is not necessary
 The CSS Layout in "Add User" window is broken
 When disabling/enabling an user, a confirmation dialog should appear
 Instead of showing the assignments in the snack bar, the assignment should be listed within the confirmation container

### PR DESCRIPTION
**Pull request by daniel-arnauer**
- [x]  It is not possible to add the permissions global.enableUser and global.disableUser in UI
- [x] Closed resources must be ignored when checking assignments (API level bug)
- [x] After canceling the window to add global permissions, the userlist re-fetches, this is not necessary
- [x] The CSS Layout in "Add User" window is broken
- [x] The user permissions of disabled user should be possible to edit
- [x] When disabling/enabling an user, a confirmation dialog should appear
- [x] Instead of showing the assignments in the snack bar, the assignment should be listed within the confirmation container

Closes #536

### CODE REVIEW

File: api/src/user_permissions_list.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export function addHttpHandler(server: FastifyInstance, urlPrefix: string, servi
         const userPermissionsResult = await service.getUserPermissions(ctx, user, userId);
 
         if (Result.isErr(userPermissionsResult)) {
-          throw new VError(userPermissionsResult, "user.intent.revokePermission failed");
+          throw new VError(userPermissionsResult, "user.intent.listPermission failed");
```
Comment (daniel-arnauer) [inline api/src/user_permissions_list.ts:89]: Correction of previous refactor PR

File: frontend/src/pages/Common/CreationDialog.js [hunk 2]
```diff
@@ -13,7 +13,11 @@ const styles = {
   paperRoot: {
     width: "100%",
     overflow: "scrollable"
-  }
+  },
+  container: {},
```
Comment (Stezido) [inline frontend/src/pages/Common/CreationDialog.js:17]: remove empty styles

File: frontend/src/pages/Users/DisableUserDialog.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
+import { Typography } from "@material-ui/core";
```
Comment (Stezido) [inline frontend/src/pages/Users/DisableUserDialog.js:1]: Please use the ConfirmationDialog component

File: frontend/src/pages/Users/EnableUserDialog.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
+import { Typography } from "@material-ui/core";
```
Comment (Stezido) [inline frontend/src/pages/Users/EnableUserDialog.js:1]: Please use the ConfirmationDialog component

File: frontend/src/pages/Users/PasswordDialog.js [hunk 5]
```diff
@@ -18,7 +18,11 @@ const styles = {
   paperRoot: {
     width: "100%",
     overflow: "scrollable"
-  }
+  },
+  container: {},
```
Comment (Stezido) [inline frontend/src/pages/Users/PasswordDialog.js:22]: remove empty styles

File: frontend/src/pages/Users/UserDialogContent.js [hunk 6]
```diff
@@ -21,28 +21,22 @@ const styles = {
...
   infoIcon: {
     fontSize: 20,
     marginRight: "10px"
   },
   info: {
     display: "flex",
     paddingRight: 20
-  }
+  },
+  customWidth: {},
```
Comment (Stezido) [inline frontend/src/pages/Users/UserDialogContent.js:37]: remove empty styles

File: frontend/src/pages/Users/actions.js [hunk 7]
```diff
@@ -70,6 +70,15 @@ export const ENABLE_USER_SUCCESS = "ENABLE_USER_SUCCESS";
 export const DISABLE_USER = " DISABLE_USER";
 export const DISABLE_USER_SUCCESS = " DISABLE_USER_SUCCESS";
 
+export const SHOW_ENABLE_DIALOG = "SHOW_ENABLE_DIALOG";
```
Comment (Stezido) [inline frontend/src/pages/Users/actions.js:73]: not needed if using the existing confirmation dialog

**Review by Stezido — CHANGES_REQUESTED** [review]
Is an own endpoint for listAssigments needed? 
Suggestion: Open up the confirmation dialog when the disableUser saga is called. After confirmation the frontend tries to disable the user, but get an error which leads to a filled assignments table in the confirmation dialog.
When the new endpoint should be kept the permission for the intent listAssignments should be the intent global.listAssignments not global.disableUser. If the issuer has the global.listAssignments permission but cannot view the projects, subproject or workflowitem the api cannot give information about where the user is assigned to.
Use the existing ConfirmationDialog.

File: api/src/service/user_assignments_get.ts [hunk 8]
```diff
@@ -3,20 +3,24 @@ import { VError } from "verror";
...
+import { getGlobalPermissions } from "./global_permissions_get";
+import { ServiceUser } from "./domain/organization/service_user";
 
 export async function getUserAssignments(
   conn: ConnToken,
   ctx: Ctx,
-  userId: UserDisable.RequestData,
+  issuer: ServiceUser,
+  issuerOrganization: string,
+  user: UserAssignmentsGet.RequestData,
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply api/src/service/user_assignments_get.ts:16]: done

File: frontend/src/pages/Confirmation/ConfirmationContainer.js [hunk 9]
```diff
@@ -34,6 +34,9 @@ import { applyOriginalActions, createAdditionalActions } from "./createAdditiona
 import { executeOriginalActions } from "./executeOriginalActions";
 
 class ConfirmationContainer extends React.Component {
+  componentDidMount() {
+    this.props.cancelConfirmation(undefined);
```
Comment (Stezido) [inline frontend/src/pages/Confirmation/ConfirmationContainer.js:38]: Why this is needed?

File: frontend/src/pages/Users/DisableUserDialogContent.js [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+    }
+    if (assignmentType === "workflowitem") {
+      url = `${baseurl}/projects/${assignment.projectId}/${assignment.subprojectId}`;
+    }
+    return (
+      <div key={assignment.id}>
+        <a href={url} target="_blank" rel="noopener noreferrer">
+          {assignment.displayName}
+        </a>
+        <br />
```
Comment (Stezido) [inline frontend/src/pages/Users/DisableUserDialogContent.js:69]: use css styles instead of br tags

**Review by Stezido — CHANGES_REQUESTED** [review]
There is a console warning in frontend when creating a new group.
when canceling the createUser/createGroup dialog listpermissions are fetched and the site is rerendered. If the dialog is canceled nothing should be fetched

File: frontend/src/pages/Confirmation/ConfirmationContainer.js (hunk 9 above)
Comment (daniel-arnauer) [reply frontend/src/pages/Confirmation/ConfirmationContainer.js:38]: This is needed to reset the redux state of Confirmation Dialog component. Bugs occurred after pressing the submit button and the API rejects the request - in this case, the submit-state remains true in redux store, hence the dialog is not showing up anymore (and the request are send immediately - without confirmation)

Solution: `cancelConfirmation()` only sets defaultState. For better readability, I should create another function called `resetConfirmation()` that also set the redux state to defaultState.
Another solution: Add action `CONFIRMATION_UNSUCESSFULL `and yield it on every API error related to confirmation dialog. This action resets redux state of confirmation dialog like this `state.set("confirmed", defaultState.get("confirmed"));`

File: frontend/src/pages/Users/DisableUserDialogContent.js (hunk 10 above)
Comment (daniel-arnauer) [reply frontend/src/pages/Users/DisableUserDialogContent.js:69]: done

File: e2e-test/cypress/integration/disable_enable_user_spec.js [hunk 11]
```diff
@@ -172,31 +195,29 @@ describe("Login", function() {
...
-      // Disable user
+      // Open disable dialog
       cy.get(`[data-test=disable-user-${testUserId}]`)
         .should("be.visible")
         .click();
-      cy.wait("@disableUser").then(xhr => {
-        expect(xhr.response.body.error.code).to.eql(412);
-      });
+      cy.get(`[data-test=confirmation-dialog-confirm]`).should("be.disabled");
+      cy.wait("@fetchAssignments");
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply e2e-test/cypress/integration/disable_enable_user_spec.js:203]: I change the position

File: e2e-test/cypress/integration/disable_enable_user_spec.js [hunk 12]
```diff
@@ -172,31 +195,29 @@ describe("Login", function() {
...
-        // Disable user
+        // Open disable dialog
         cy.get(`[data-test=disable-user-${testUserId}]`)
           .should("be.visible")
           .click();
-        cy.wait("@disableUser").then(xhr => {
-          expect(xhr.response.body.error.code).to.eql(412);
-        });
+        cy.get(`[data-test=confirmation-dialog-confirm]`).should("be.disabled");
+        cy.wait("@fetchAssignments");
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply e2e-test/cypress/integration/disable_enable_user_spec.js:220]: I change the position

File: api/src/service/domain/workflow/user_assignments_get.ts [hunk 13]
```diff
@@ -29,52 +78,67 @@ export async function getUserAssignments(
...
     const subprojects = await repository.getSubprojects(project.id);
     if (Result.isErr(subprojects)) {
       return new VError(subprojects, "failed to fetch subprojects");
     }
+
     for await (const subproject of subprojects) {
       if (subproject.status === "closed") continue;
       if (subproject.assignee === userId) {
-        assignedSubprojects.push(subproject);
+        if (!isRoot && !Subproject.permits(subproject, issuer, subprojectIntents)) {
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply api/src/service/domain/workflow/user_assignments_get.ts:97]: Here, this checks if the issuer is the root-user

**Review by Stezido — APPROVED** [review]

**Pull request merged**